app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,133 @@
+    var variant = this.module.getAttribute('data-ecommerce-variant') || undefined
+    var ecommerceRows = this.module.querySelectorAll('[data-ecommerce-row]')
+    var listTitle = this.module.getAttribute('data-list-title') || this.DEFAULT_LIST_TITLE
+    var startPosition = parseInt(this.module.getAttribute('data-ecommerce-start-index'), 10)
+    var ecommerceObject = {}
+
+    ecommerceObject.event = 'search_results'
+    ecommerceObject.search_results = {}
+    ecommerceObject.search_results.event_name = searchResultClicked && searchResult ? 'select_item' : 'view_item_list'
+    ecommerceObject.search_results.term = searchQuery
+    ecommerceObject.search_results.sort = variant
+    ecommerceObject.search_results.results = parseInt(this.getMetaTagContent('govuk:search-result-count'))
+    ecommerceObject.search_results.ecommerce = {}
+    ecommerceObject.search_results.ecommerce.items = []
+
+    if (searchResultClicked && searchResult) {
+      ecommerceObject.search_results.ecommerce.items.push({
+        item_id: searchResult.getAttribute('data-ecommerce-path'),
+        item_name: searchResult.textContent,
+        item_list_name: listTitle,
+        index: this.getIndex(searchResult, Array.prototype.indexOf.call(ecommerceRows, searchResult)) + startPosition
+      })
+
+      window.GOVUK.analyticsGA4.analyticsModules.Ga4LinkTracker.internalLinksDomain = 'www.gov.uk/'
+      window.GOVUK.analyticsGA4.analyticsModules.Ga4LinkTracker.internalLinksDomainWithoutWww = 'gov.uk/'
+
+      ecommerceObject.event_data = {
+        external: window.GOVUK.analyticsGA4.analyticsModules.Ga4LinkTracker.isExternalLink(searchResult.getAttribute('data-ecommerce-path')) ? 'true' : 'false'
+      }
+    } else {
+      for (var i = 0; i < ecommerceRows.length; i++) {
+        var ecommerceRow = ecommerceRows[i]
+        var path = ecommerceRow.getAttribute('data-ecommerce-path')
+
+        ecommerceObject.search_results.ecommerce.items.push({
+          item_id: path,
+          item_list_name: listTitle,
+          index: this.getIndex(ecommerceRow, i) + startPosition
+        })
+      }
+    }
+
+    return ecommerceObject
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.getIndex = function (element, index) {
+    var indexOverride = element.getAttribute('data-ecommerce-index')
+    return indexOverride ? parseInt(indexOverride, 10) - 1 : index
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.clearPreviousEcommerceObject = function () {
+    window.dataLayer.push({ search_results: { ecommerce: null } })
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.getMetaTagContent = function (metaTagName) {
+    var element = document.querySelector(`meta[name='${metaTagName}']`)
+    if (!element) {
+      return null;    
+    }
+    return element.getAttribute('content');
+  }
+
+  Modules.Ga4EnhancedEcommerceTracker = Ga4EnhancedEcommerceTracker
```

Comment:
Since this gets referred to in a few places, would it be cleaner to call this something shorter? Maybe just `Ga4EcommerceTracker`?

Reply:
Agreed. As per our discussion earlier, this will be addressed as part of a wider task to look at naming conventions.